actors/miner/src/lib.rs

```diff
@@ -1764,27 +1771,28 @@ impl Actor {
 
         // Validate pre-commits.
         let allow_deals = false; // New onboarding entry point does not allow pre-committed deals.
-        let (batch_return, proof_inputs) =
+        let (validation_batch, proof_inputs) =
             validate_precommits(rt, &precommits, allow_deals, params.require_activation_success)?;
-        if batch_return.success_count == 0 {
+        if validation_batch.success_count == 0 {
             return Err(actor_error!(illegal_argument, "no valid precommits specified"));
         }
-        let valid_precommits = batch_return.successes(&precommits);
-        let valid_activation_inputs = batch_return.successes(&params.sector_activations);
+        let valid_precommits = validation_batch.successes(&precommits);
+        let valid_activation_inputs = validation_batch.successes(&params.sector_activations);
         let eligible_activation_inputs_iter = valid_activation_inputs.iter().zip(valid_precommits);
 
         // Verify seal proof(s), either batch or aggregate.
         let mut proven_activation_inputs: Vec<(
             &SectorActivationManifest,
             &SectorPreCommitOnChainInfo,
         )> = vec![];
+        let mut proven_batch_gen = BatchReturnGen::new(validation_batch.size());
         if !params.sector_proofs.is_empty() {
             // Verify batched proofs.
             // Filter proof inputs to those for valid pre-commits.
-            let seal_verify_inputs: Vec<SealVerifyInfo> = batch_return
```

Comment:
Same mild suggestion about separating this into a function and then using the `successes` filter over loop inputs to generate `proven_activation_inputs`

Reply:
Noted for a follow-up